[19.0][REF] sale_order_type: replace analytic_account_id by analytic distribution

lef-adhoc opened the pull request:
The analytic_account_id field was defined in the sale.order.type model but was not being used anywhere in the module's logic. This field has no practical functionality in the current implementation.

Additionally, this field was removed from the standard Odoo sale.order model in version 18.0

lef-adhoc committed:
[REF] sale_order_type: replace analytic_account_id by analytic distribution

lef-adhoc commented:
@pedrobaeza Could you check this?

pedrobaeza commented:
Well, I think the goal is to have a default analytic mapping, but as you say, right now it hasn't been covered.

matiasperalta1 reviewed (approved):
LGTM

alexey-pelykh reviewed (approved):
Clean removal of dead code. I verified that `analytic_account_id` is defined on `sale.order.type` but never consumed anywhere in the module's logic -- no compute, no `_prepare_invoice`, no tests, no onchange references it. The field was purely decorative (visible in the form but never wired to anything). The PR description correctly notes this field was also dropped from standard `sale.order` in Odoo 18.0.

CI is fully green. LGTM.

rousseldenis commented:
In fact, it has been removed partially in 18.0 migration following : https://github.com/OCA/sale-workflow/pull/3352#discussion_r1814369380

IMHO, it should be replaced by a default analytic distribution.

lef-adhoc commented:
@rousseldenis @pedrobaeza What do you think of the change? (Please ignore the version change in the manifest – this is because we’ve included the PR and need to update the module; I’ll delete it before it gets merged.)

pedrobaeza reviewed (commented):
You should include a migration script to convert the analytic account into the analytic distribution.

lef-adhoc commented:
@pedrobaeza I added the migration script there – how does it look now?

pedrobaeza commented on sale_order_type/migrations/19.0.1.1.0/post-migration.py:
```diff
@@ -0,0 +1,30 @@
+        """
+    )
+
+    for record_id, analytic_account_id in cr.fetchall():
```
Don't iterate on this, but do it in one shot with a single SQL. You can build a JSON in SQL.

lef-adhoc replied:
@pedrobaeza Done!

pedrobaeza commented on sale_order_type/migrations/19.0.1.3.0/post-migration.py:
```diff
@@ -0,0 +1,17 @@
+from openupgradelib import openupgrade
+
+
+@openupgrade.migrate()
```
As you are not using anything of `openupgradelib` here, you can remove it and us raw `cr`.

pedrobaeza commented on sale_order_type/migrations/19.0.1.3.0/post-migration.py:
```diff
@@ -0,0 +1,17 @@
+@openupgrade.migrate()
+def migrate(env, version):
+    cr = env.cr
+
```
Better to not introduce empty lines inside a method.

lef-adhoc commented:
@pedrobaeza Done! Removed [openupgradelib](vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/workbench.html) and used raw [cr](vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/workbench.html) with a plain SQL check, and removed the empty lines inside the method.

pedrobaeza commented on sale_order_type/migrations/19.0.1.3.0/post-migration.py:
```diff
@@ -0,0 +1,14 @@
+def migrate(cr, version):
+    cr.execute(
```
You can use here `tools.column_exists` in Odoo for commodity, but let's merge anyway,as this is correct.

pedrobaeza reviewed (approved):
/ocabot merge nobump
